### ISSUE TITLE
[css-animations-2][scroll-animations-1] Add scroll() as value for animation-timeline

### ISSUE THREAD

**Issue by ydaniv** [w3c/csswg-drafts#7579]
According to [this](https://drafts.csswg.org/scroll-animations/#scroll-notation) the `animation-timeline` should accept the new `scroll()` function as a value, but this is still missing from the [animations spec](https://drafts.csswg.org/css-animations-2/#animation-timeline) definition.

And speaking of which, should we have a corresponding `view()` function for `ViewTimeline` for same purpose?

**Comment by SebastianZ** [comment]
But it is defined in the Scroll-linked Animations spec. So why should it be moved to the CSS Animations 2 spec.?

Regarding `view()`, I'd say yes, at first sight, though that should be discussed in a separate issue.

Sebastian

**Comment by ydaniv** [comment]
@SebastianZ 

> So why should it be moved to the CSS Animations 2 spec.?

Not moved there, but added as a  possible value.

Currently `animation-timline` can take `<single-animation-tineline>` which is defined as:
```
auto | none | <<timeline-name>>
```
And `<<timeline-name>>` is defined as:
```
<<custom-ident>> | <<string>>
```
IIUC these don't include the `scroll()` function defined in scroll-animations, so just need to add it.

**Comment by SebastianZ** [comment]
> > So why should it be moved to the CSS Animations 2 spec.?
> 
> Not moved there, but added as a possible value.

If it's added there, it should still refer to the definition in Scroll-linked Animations to avoid double-specification.
Though note that there are several specifications that extend existing property syntaxes. Examples are the [logical values for `float` and `clear` in CSS Logical 1](https://drafts.csswg.org/css-logical-1/#float-clear), [`running()` for `position` in CSS GCPM 3](https://drafts.csswg.org/css-gcpm-3/#running-syntax) or [`min-content`, `max-content` and `fit-content()` for `column-width` in CSS Sizing 3](https://drafts.csswg.org/css-sizing-3/#column-sizing). So it is not _required_ to add it there but it _could_ be added there for clarity.

Sebastian

**Comment by ydaniv** [comment]
I see, well that's interesting to know - not an editor myself of neither.

I guess form a reader's POV it would be nice to have a reference in css-animations-2 refer to the definition in scroll-animations, but again, not spec expert yet  (:

**Comment by fantasai** [comment]
I clarified the integration of these functions into animation-timeline in https://github.com/w3c/csswg-drafts/commit/6c958c3ea8d8e9750bd556df9b8b2bde261e564c

And I made a pull request against CSS Animations 2 to integrate into its definition of `animation-timeline` in https://github.com/w3c/csswg-drafts/pull/8199

This also raised the question of how scroll/view progress timelines are prioritized in the named timeline lookup... I thought it made the most sense to look them up first, before at-rule defined timelines. But open to other ideas. Lmk. CC @flackr
